lms/djangoapps/teams/static/teams/js/spec/edit_team_spec.js

```diff
@@ -0,0 +1,189 @@
+
+    describe('EditTeam', function () {
+        var teamEditView,
+            teamsUrl = '/api/team/v0/teams/',
+            teamsData = {
+                id: null,
+                name: "TeamName",
+                is_active: null,
+                course_id: "a/b/c",
+                topic_id: "awesomeness",
+                date_created: "",
+                description: "TeamDescription",
+                country: "c",
+                language: "a",
+                membership: []
+            },
+            verifyValidation = function (requests, fieldsData) {
+                _.each(fieldsData, function (fieldData) {
+                    teamEditView.$(fieldData[0]).val(fieldData[1]);
+                });
+
+                teamEditView.$('.create-team.form-actions .action-primary').click();
+
+                var message = teamEditView.$('.wrapper-msg');
+                expect(message.is(':visible')).toBeTruthy();
+                expect(message.find('.title').text().trim()).toBe("Your team could not created!");
+                expect(message.find('.copy').text().trim()).toBe(
+                    "Check the highlighted fields below and try again."
+                );
+
+                _.each(fieldsData, function (fieldData) {
+                    if(fieldData[2] === 'error') {
+                        expect(teamEditView.$(fieldData[0].split(" ")[0] + '.error').length).toBe(1);
+                    } else if(fieldData[2] === 'success') {
+                        expect(teamEditView.$(fieldData[0].split(" ")[0] + '.error').length).toBe(0);
+                    }
+                });
+
+                expect(requests.length).toBe(0);
+            },
+            expectContent = function (selector, text) {
+                expect(teamEditView.$(selector).text().trim()).toBe(text);
+            },
+            verifyDropdownData = function (selector, expectedItems) {
+                var options = teamEditView.$(selector)[0].options;
+                var renderedItems = $.map(options, function( elem ) {
+                    return [[elem.value, elem.text]];
+                });
+                for (var i = 0; i < expectedItems.length; i++) {
+                    expect(renderedItems).toContain(expectedItems[i]);
+                }
+            };
+
+        beforeEach(function () {
+            setFixtures('<div class="teams-content"></div>');
+            spyOn(Backbone.history, 'navigate');
+            teamEditView = new TeamEditView({
+                el: $('.teams-content'),
+                teamParams: {
+                    teamsUrl: teamsUrl,
+                    courseId: "a/b/c",
+                    topicId: 'awesomeness',
+                    topicName: 'Awesomeness',
+                    languages: [['a', 'aaa'], ['b', 'bbb']],
+                    countries: [['c', 'ccc'], ['d', 'ddd']]
+                }
+            }).render();
+        });
+
+        it('can render itself correctly', function () {
```

Comment:
I see that you've moved the whole test case into `edit_team_spec.js`. This isn't what I meant by my original comment (shown below). In particular, you don't need to test the existence of every single DOM element of the view, since that stuff might change in the future. There also needs to be a spec for the Create Team view. Please add that back.
* For the create team spec:
    * Test that the header is rendered by checking for the existence of its top-level DOM element
    * Test that the edit view is rendered by checking for the existence of its top-level DOM element
* For the edit team spec:
    * Test that all the fields have been rendered, but don't test everything about the fields - they're tested in their own specs, right?

> Wow! You've really verified every single component of this view. I like that your test case is comprehensive, but I'm concerned that this is going to be a really brittle test case. Can you do a couple things to clean it up?
>
> * We already test the header view in header_spec.js, so I think you should only check that the header appears.
> * Move the logic that checks form rendering into edit_team_spec.js, since it has to do with the edit team view.
>     * After you've moved that logic, you can simply here test that the form is present.